python/dcr_template.py

```diff
@@ -132,78 +195,154 @@ def generate_templates_from_model(self, obsid_range=None, exposures=None, add_no
             else:
                 obsid = self._fetch_metadata(calexp.getMetadata(), "OBSID", default_value=0)
```

Comment:
I've found ways to eliminate all uses of `_fetch_metadata`, either by deriving the quantities or requiring them to be supplied by the user.